zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -139,9 +138,6 @@ impl ExecutionState {
             Self::ErrorInvalidOpcode
                 | Self::ErrorStack
                 | Self::ErrorWriteProtection
-                | Self::ErrorDepth
-                | Self::ErrorInsufficientBalance
-                | Self::ErrorContractAddressCollision
```

Comment:
@lispc Will this make thing worse if we remove them before implementing them in `CALL` or `CREATE`?

Reply:
`ErrorDepth`, `ErrorContractAddressCollision`, `ErrorInsufficientBalance` these few states only use along with `DummyGadget`, 
 and these states never touch `halts_in_exception` because halts_in_exception only use in `RestoreContextGadget`. Therefore its constraints never activated. Maybe we open another issue to tracking the implemetatoin~?

Reply:
Make sense, for `ErrorContractAddressCollision` we got https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1309 and for `ErrorDepth` we have https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/1312 now.